common/src/stack/report-system/command/report/system/tests/test_partitions.py

```diff
@@ -0,0 +1,129 @@
+from stack.commands.report.system.fixtures.disk_partition_check import get_part_size, get_part_fs
+
+testinfra_hosts = list(set([host['host'] for host in api.Call('list host partition')]))
+class TestStoragePartition:
+
+	""" Test if the way stacki configured the disks is still how they are partitioned """
+
+	def test_storage_partition(self, host, get_partitions, get_part_label, get_part_size, get_part_fs):
+
+		# Get current hostname and test if we can ssh into the host
+		# Otherwise fail the test
+		try:
+			hostname = host.check_output('hostname')
+
+		except paramiko.ssh_exception.NoValidConnectionsError:
+			pytest.fail(f'Could not ssh into host')
+
+		# Get stacki storage config using proper scoping
+		storage_config = api.Call(f'list host storage partition {hostname}')
+		check_config = defaultdict(dict)
+		partition_info = namedtuple('partition', ['name', 'mountpoint', 'label', 'fstype', 'size'])
+		errors = []
+
+		# Otherwise if we are on an older version of stacki without proper scoping,
+		# use report host instead and convert the dict output as a string into an actual dict
+		if not storage_config:
+			report_storage = api.Call(f'report host storage partition {hostname}')
+
+			# If we still cannot get stacki's storage config for the current host,
+			# fail the test
+			if not report_storage:
+				pytest.fail(f'Could not get stacki partition config for host {hostname}')
+			else:
+				storage_config = json.loads(report_storage[0].replace("'", '"').replace('None', '""'))
+
+		# Get a list of the partitions that are actually on the disk
+		host_partitions = get_partitions(host)
+
+		# Go through the stacki storage config and see
+		# if it matches what's actually on the disk/disks
+		for partition in storage_config:
+			# Info needed for each partiton
+			disk = partition['device']
+			disk_num = str(partition['partid'])
+			partition_name = disk + disk_num
+			conf_mntpt = partition['mountpoint']
+			part_label = ''
+
+			# If a partition cannot be found on the actual disk, return an error
+			if partition_name not in host_partitions:
+				errors.append(f'/dev/{partition_name} not found on host')
+				continue
+
+			# Because testinfra only has mountpoint based partition functions
+			# use them when we can but if there is a partition without a mountpoint
+			# check using a fixture that directly calls lsblk
+			if conf_mntpt != (None or "") :
+
+				if not host.mount_point(conf_mntpt).exists:
+					errors.append(f'Could not find {conf_mntpt} on disk')
+
+				else:
+					curr_mnt_part = host.mount_point(conf_mntpt).device
+					curr_fs = host.mount_point(conf_mntpt).filesystem
+
+					if curr_mnt_part != f'/dev/{partition_name}':
+						msg = f'Mountpoint {conf_mntpt} found at {curr_mnt_part} but was configured to be at /dev/{partition_name}'
+						errors.append(msg)
+
+					if (curr_fs != partition['fstype']) and (partition['fstype'] != None):
```

Comment:
Checking for `None` should use `is` or `is not` instead of `==` or `!=`. `None` is a special object such that all uses of `None` refer to the same object, so `is` is more correct here and is the "pythonic" way.

Something something evil people could make custom objects compare equal/not equal to `None`, but it's much harder to make something have the same identity as `None`.